### DISCUSSION TITLE
修改短标题脚本但弹窗两次

### DISCUSSION THREAD

**Discussion by pencilheart** [windingwind/zotero-actions-tags#412] [Q&A] [answered]
您好，想写一个脚本使用快捷键弹出窗口进行短标题的修改，遇到一个问题：点击ok或cancel后再次出现输入窗口。如何取消第二次的输入弹窗，请求支援🙏
目前完成的代码如下。
```js
var zoteroPane = Zotero.getActiveZoteroPane();
var items = zoteroPane.getSelectedItems();

for (let item of items) {
    if (item && !item.isNote() && item.isRegularItem()) { // 正常条目，非笔记
        var currentShortTitle = item.getField("shortTitle"); // 获取当前短标题

        // 弹出对话框，要求用户输入新的短标题，当前短标题作为默认值
        var newShortTitle = require('window').prompt("编辑短标题（当前: " + currentShortTitle + "）", currentShortTitle);

        // 如果用户点击取消或未输入新短标题，则跳过该条目
        if (newShortTitle !== null) {
            item.setField("shortTitle", newShortTitle);
            await item.saveTx(); // 保存更改
        }
        
    }
}
```

https://github.com/user-attachments/assets/6dbdc3d3-8d40-46fe-8a9d-dafc9060d134

**Comment by windingwind** [comment, marked as answer]
https://github.com/windingwind/zotero-actions-tags?tab=readme-ov-file#custom-script

> When selecting multiple items in the library, the action will be triggered once for all items (items=[...], item=undefined) and then one by one for each item (items=[], item=...). You can use the items variable to get the selected items array and avoid duplicate executions.
